File: GodsOfWargameBrowser/src/main/webapp/MovementScripts/Routing.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

function Routing(){
    this.routing = [];
    
}
function addMove(coordinate){
    Routing.routing.push(coordinate);
}
function createMovementArray(){
    var movement = new Array();
    for ( let i=0;i<TerrainMap.length;i++){
        movement[i] = new Array();
    }
}
//converts Terrain strings to Ints
function convertTerrainToInt(unit){
    //Create 2d array
    let movement = new Array();
    for ( let i=0;i<arrayHeight;i++){
        movement[i] = new Array();
        for(let j = 0; j< arrayWidth;j++){
            movement[i][j] = 0;
        }
    }
    //
    for(let i=0;i< TerrainMap.length;i++){
        for(j=0; j<unit.terrainRules.badSpots.length; j++){
            //Set a default value of walkable before deciding if possible
          //movement[TerrainMap[i].yVal][TerrainMap[i].xVal] = 0;
            
            if( TerrainMap[i].type === unit.terrainRules.badSpots[j]) {
                //Set movement to 1 to indicate no walkable path
                movement[TerrainMap[i].yVal][TerrainMap[i].xVal] = 1;
                //leave for loop as only 1 badSpot is needed
                break;
            } 
        }
        
    }
    console.log("movement array");
    console.log(movement);
    return movement;
}

File: GodsOfWargameBrowser/src/main/java/Units/pathing/Routing.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Units.pathing;

import Location.Coordinate;
import java.util.ArrayList;
import java.util.List;

/**A class that represents the path that will be taken by the unit traveling.
 * For performance reasons as the unit moves it should remove it's prior location from route.
 * Failure to do so may open up pathing failures should old terrain transform.
 * Alternatives to removal would be to keep an index of cycles so that old squares are ignored
 * Uses ArrayList as opposed to Set so that the unit can loop back around and re-traverse land
 * @author brenn
 */
public class Routing {
    //public List<Coordinate> pathingRoute = new ArrayList<>();
    public List<int[]> pathingRoute = new ArrayList<>();

    public List<int[]> getPathingRoute() {
        return pathingRoute;
    }

    public void setPathingRoute(ArrayList<int[]> newRoute){
        pathingRoute = newRoute;
    }
    
}
